interview-questions.md

```diff
@@ -8,36 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: In this case you would have to generate a new migration file that will add a cohort column to the sudents table. The foreign key is going to be cohort_id since the students are going to have association with the Cohort model. The foreign key only has to be designated to the Student model since the student table will be referencing the cohort table.
 
 Researched answer:
+First you run the rail g migration AddCohortRefToStudents cohort:references to establish the realationship between the two tables and add the foreign key. The foreign key will belong to the students table. Once the migration is generates you will need to run a raild db:migrate command to update the database schema. Once that is complete we check that the Cohort model has an association of has_many and the Student model has belongs_to that will use the foreing key to associate.
+
+Used class notes.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: There are 4 RESTful routes that must be passed params. Those would be show, update, and destroy in the case of these it would be because there is a request to show, update or delete a specific entry which would requre an :id to be passed to locate the exact part of the model. The last one which is create that is passed the private params that can only be called by the helper methods and not the user this would be the case for when and entry to the model is going to be created we want to make sure that the data entry is going to be a valid one so it has the system check against the paramaters that are set to deem the entry valid.
 
-Researched answer:
+Researched answer: Altough there us no RESTful route that requires parameters there are some that are commonly used. Depending what your app is trying to accomplish the rotues that require parameters may vary. The most commonly used would be Query parameters(GET), Path parameters (GET/DELETE) and Body query(POST/PUT). In all of these three cases the user would be requesting either a single peice of data to view, update or create which would make the most sense to passa a paramter which is typically the id or a unique identifier of some sort.
```

Comment:
Yep, this is a strange question for sure. Your original answer is what the question was looking for, since as far as the scope of this course, we are using show, edit, update, destroy with params!